rfcs/2021-05-17-7469-load-balancing-k8s-aggregators.md

```diff
@@ -0,0 +1,110 @@
+# RFC 7469 - 2021-05-16 - Scaling and load balancing for Vector aggregators in Kubernetes
+
+This RFC describes the need for a tooling and environment agnostic load balancing solution to be bundled with the Vector aggregator Helm chart.
+
+* [Scope](#scope)
+* [Out of Scope](#out-of-scope)
+* [Motivation](#motivation)
+* [Internal Proposal](#internal-proposal)
+* [Rationale](#rationale)
+* [Prior Art](#prior-art)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+  * [Do Nothing](#do-nothing)
+  * [Only Client-side Load Balancing](#only-client-side-load-balancing)
+  * [Require a Service Mesh](#require-a-service-mesh)
+* [Outstanding Questions](#outstanding-questions)
+* [Plan Of Attack](#plan-of-attack)
+
+## Scope
+
+Load balancing will be a concern for any `sink` or `source` supported by Vector; some of which can use a general solution (ex: load balancing for our HTTP based `sinks`) and some of which are specific to the component (ex: Kafka `source`/`sink`). Due to the breadth of the topic, this RFC will focus on three specific cases for load balancing to Vector aggregators in Kubernetes, while also giving consideration to future adoption for other components.
+
+* Vector agent to Vector aggregator
+* Datadog agent to Vector aggregator
+* Syslog (TCP) agent to Vector aggregator
+
+## Out of Scope
+
+* Scaling the Kafka `source`
+* Scaling and load balancing on platforms other than Kubernetes
+* Load balancing UDP based `sources`
```

Comment:
I think this is _in_ scope, but just an aspect that we will be deferring due technical limitations.

Reply:
I might have misunderstood - is it out of scope for the document or for the work being done?

Reply:
It's just a technical limitation that we're willing to accept for the initial implementation, but it is not out of scope for this project. If we can identify a similarly simple solution then we should consider it.